### PR TITLE
Add integration tests for the RAG pipeline covering OpenAI and Bedrock

### PR DESCRIPTION
**Pull request by austintlee**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by austintlee** [comment]
Also fixes https://github.com/opensearch-project/ml-commons/issues/2169

**Comment by austintlee** [comment]
@dhrubo-os @ylwu-amzn can you guys take a look?  Thanks!

**Comment by ylwu-amzn** [comment]
```
> Task :opensearch-ml-search-processors:compileTestJava FAILED
/__w/ml-commons/ml-commons/search-processors/src/test/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeQAResponseProcessorTests.java:470: error: no suitable constructor found for GenerativeQAParameters(String,String,String,int,<null>,<null>)
        GenerativeQAParameters params = new GenerativeQAParameters("12345", "llm_model", "You are kind.", contextSize, null, null);
```

**Comment by austintlee** [comment]
@ylwu-amzn Can you take a look again?

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+            .makeRequest(
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\"plugins.ml_commons.memory_feature_enabled\":true}}",
+                ImmutableList.of(new BasicHeader(HttpHeaders.USER_AGENT, DEFAULT_USER_AGENT))
+            );
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        */
```
Comment (dhrubo-os) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java:233]: Should we remove this?

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+                client(),
+                "PUT",
+                "_cluster/settings",
+                null,
+                "{\"persistent\":{\n"
+                    + "        \"plugins.ml_commons.trusted_connector_endpoints_regex\": [\n"
+                    + "            \"^https://runtime\\\\.sagemaker\\\\..*\\\\.amazonaws\\\\.com/.*$\",\n"
+                    + "            \"^https://api\\\\.openai\\\\.com/.*$\",\n"
+                    + "            \"^https://api\\\\.cohere\\\\.ai/.*$\",\n"
+                    + "            \"^http://127.0.0.1:1323/.*$\",\n"
```
Comment (dhrubo-os) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java:246]: why do we need this?

**Review by dhrubo-os — APPROVED** [review]

**Comment by dhrubo-os** [comment]
2 minor comments, approving the PR.

**Review by HenryL27 — APPROVED** [review]

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java (hunk 1 above)
Comment (austintlee) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java]: Removed.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java (hunk 2 above)
Comment (austintlee) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLRAGSearchProcessorIT.java:222]: I must have used it for some local testing.  I will remove it.

**Comment by austintlee** [comment]
```
REPRODUCE WITH: gradlew ':opensearch-ml-plugin:test' --tests "org.opensearch.ml.action.prediction.PredictionITTests.testPredictionWithEmptyDataset_KMeans" -Dtests.seed=9E0AF175677EECFB -Dtests.security.manager=false -Dtests.locale=de-AT -Dtests.timezone=AGT -Druntime.java=21
org.opensearch.ml.action.prediction.PredictionITTests > testPredictionWithEmptyDataset_KMeans FAILED
    java.lang.AssertionError: 
    Expected: (an instance of java.lang.IllegalArgumentException and exception with message a string containing "No document found")
         but: an instance of java.lang.IllegalArgumentException <org.junit.internal.runners.model.MultipleFailureException: There were 2 errors:
      java.lang.IllegalArgumentException(No document found)
      java.lang.AssertionError(Shard [.plugins-ml-config][0] is still locked after 5 sec waiting)> is a org.junit.internal.runners.model.MultipleFailureException
    Stacktrace was: java.lang.IllegalArgumentException: No document found
    	at org.opensearch.ml.engine.indices.MLInputDatasetHandler.lambda$parseSearchQueryInput$0(MLInputDatasetHandler.java:57)
```

I don't think that's related to my change.

**Review by HenryL27 — APPROVED** [review]
good ol' KMeans prediction test failure

**Comment by austintlee** [comment]
@ylwu-amzn can you merge this?

**Review by dhrubo-os — APPROVED** [review]

**Comment by dhrubo-os** [comment]
backport PRs are failing. Could you please check? @austintlee

**Comment by austintlee** [comment]
https://github.com/opensearch-project/ml-commons/pull/2235 @dhrubo-os does this look ok?  The diff doesn't look right to me.

**Comment by austintlee** [comment]
`git worktree add .worktrees/backport-2.13 2.13` is resulting in `fatal: invalid reference: 2.13`.

**Comment by austintlee** [comment]
oh it seems `2.13` has not been created?!

**Comment by austintlee** [comment]
For 2.x, I get this message:

```
There isn’t anything to compare.
opensearch-project:2.x is up to date with all commits from austintlee:backport/backport-2213-to-2.x. Try [switching the base](https://github.com/opensearch-project/ml-commons/compare/austintlee:backport/backport-2213-to-2.x...2.x) for your comparison.
```

**Comment by dhrubo-os** [comment]
From this workflow log: https://github.com/opensearch-project/ml-commons/actions/runs/8348785119/job/22851492548

Looks like backport is failing for :

```
https://github.com/opensearch-project/ml-commons/actions/runs/8348785119/job/22851492548
```


Did you take all the updates from the branch?

**Pull request merged**